src/k8s-engine/wsl.ts

```diff
@@ -390,15 +390,33 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     return this.#desiredPort;
   }
 
+  /**
+   * List the registered WSL2 distributions.
+   */
   protected async registeredDistros({ runningOnly = false } = {}): Promise<string[]> {
-    const args = ['--list', '--quiet'];
+    const stdout = await this.execWSL({ capture: true }, '--list', '--verbose');
+    // As wsl.exe may be localized, don't check state here.
+    const parser = /^[\s\*]+(?<name>.*?)\s+\w+\s+(?<version>\d+)\s*$/;
+
+    let result = stdout.trim()
+      .split(/[\r\n]+/)
+      .slice(1) // drop the title row
+      .map(line => line.match(parser))
+      .filter(defined)
+      .filter(result => result.groups?.version === '2')
+      .map(result => result.groups?.name)
+      .filter(defined);
```

Comment:
If you change the regex to `...(?<name>.+?)` wouldn't that remove the need to check for `defined` names?

Reply:
No, because TypeScript isn't smart enough to know that `groups` exists. The `defined` is for the case where `groups` doesn't exist.

Reply:
Yup, I found that as well once I started messing with this code.